src/backend/render/mod.rs

```diff
@@ -586,20 +642,22 @@ where
         .as_ref()
         .filter(|f| !f.is_animating())
         .is_some();
-    let (overlay_elements, overlay_popups) =
+    let overlay_elements =
         split_layer_elements(renderer, output, Layer::Overlay, exclude_workspace_overview);
 
     // overlay is above everything
-    elements.extend(overlay_popups.into_iter().map(Into::into));
-    elements.extend(overlay_elements.into_iter().map(Into::into));
-
-    let mut window_elements = if !has_fullscreen {
-        let (top_elements, top_popups) =
-            split_layer_elements(renderer, output, Layer::Top, exclude_workspace_overview);
-        elements.extend(top_popups.into_iter().map(Into::into));
-        top_elements.into_iter().map(Into::into).collect()
-    } else {
-        Vec::new()
+    elements
+        .p_elements
+        .extend(overlay_elements.p_elements.into_iter().map(Into::into));
```

Comment:
Why not `overlay_elements.join` or `chain` at least? (Which should probably still yield an `ExactSizeIterator`, which should be good for `extend`.